sdk/src/beta9/runner/endpoint.py

```diff
@@ -158,6 +120,59 @@ async def _keep_worker_alive(self):
             await asyncio.sleep(1)
 
 
+@dataclass
+class TaskLifecycleData:
+    status: TaskStatus
+    result: Any
+    override_callback_url: Optional[str] = None
+
+
+class TaskLifecycleMiddleware(BaseHTTPMiddleware):
+    async def dispatch(self, request: Request, call_next):
+        if request.url.path == "/health":
+            return await call_next(request)
+
+        task_id = request.headers.get("X-TASK-ID")
+        if not task_id:
+            raise HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail="Task ID missing")
+
+        with StdoutJsonInterceptor(task_id=task_id):
+            print(f"Received task <{task_id}>")
+            start_response = request.app.state.gateway_stub.start_task(
+                StartTaskRequest(task_id=task_id, container_id=cfg.container_id)
+            )
+            if not start_response.ok:
+                raise HTTPException(
+                    status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Failed to start task"
+                )
+
+            task_lifecycle_data = TaskLifecycleData(
+                status=TaskStatus.Complete, result=None, override_callback_url=None
+            )
+
+            try:
+                request.state.task_lifecycle_data = task_lifecycle_data
+                response = await call_next(request)
+                print(f"Task <{task_id}> finished")
```

Comment:
Now that we get the `response` back directly, we don't need to do this convoluted way to passing the result and task.status. We can make it so we can extract it directly from the response

In this case, `task_lifecycle_data.result` is just `response.body`.

Also I still don't understand how we will get the task status update for user specified asgi endpoints.